src/IPC/PipeTx.cc

```diff
@@ -1,21 +1,71 @@
 #include "IPC.h"
 
-void PipeTx::send(void) {
-    this->setupPipeTx();
+#include <sys/stat.h>   /* mkfifo() */
+#include <fcntl.h>      /* O_CREAT, O_WRONLY */
+#include <iostream>     /* std::cout std::ios seekg() */
+#include <unistd.h>     /* write() */
+#include <errno.h>      /* errno */
+#include <string.h>     /* strerror */
 
-    this->fileSizeTx();
+#define FIFO "./fifoChannel"
 
+void PipeTx::send(void) {
+    this->setupPipeTx();
+    
     this->pipeTx();
 }
 
 void PipeTx::setupPipeTx(void) {
-    ;
+    int status = mkfifo(FIFO, S_IRWXU | S_IRWXG);
```

Comment:
Why do you make 2 pipes? you make it in PipeRx and PipeTx, you just need to make it in one process and the other process can juste use open()

Reply:
Well this fixes a weird bug I had on my setup, think it shouldn't affect the process too much, since if the pipe already exists it throws an error message and then just continues executing.

Reply:
Your bug might be that the part that doesn't create the pipe can't connect because the pipe doesn't exist, top avoid this you can use access( ) that checks if the fifo is ready, I did a loop and while it is not ready I just sleep.

Reply:
Yeah I tried this is as well, didn't help. :/

My current setup (as it is now in the PR) works 100% of the time for me, the only 'issue' I have that I will need to investigate, is why I need a usleep in between setting up the tx and rx side. I'd like it to work immediately without any unnecessary waiting. 

(Ah and also, without the usleep is works 95% on my setup and 100% of the time for Karol)